server/handlers/componetgenratorhelper.go

```diff
@@ -356,67 +356,57 @@ func (h *Handler) sendEventForImport(userID uuid.UUID, provider models.Provider,
+
 	if response.EntityCount.ModelCount > 0 {
 		if response.EntityCount.CompCount == 0 && response.EntityCount.RelCount == 0 && response.EntityCount.TotalErrCount == 0 {
-			message.WriteString("Model won't be registered because there was no Component or Relationship.")
-			return message.String()
+			return "Model won't be registered because there was no Component or Relationship."
 		}
 		modelName := ModelNames(response)
 		modelWord := determinePluralWord(response.EntityCount.ModelCount, "model")
 		message.WriteString(fmt.Sprintf("Imported %s %s ", modelWord, modelName))
 	}
+
 	if response.EntityCount.CompCount > 0 || response.EntityCount.RelCount > 0 {
 		message.WriteString("(")
-	}
-	if response.EntityCount.CompCount > 0 {
-		componentWord := determinePluralWord(response.EntityCount.CompCount, "component")
-		message.WriteString(fmt.Sprintf("%d %s", response.EntityCount.CompCount, componentWord))
-	}
-	if response.EntityCount.RelCount > 0 && response.EntityCount.CompCount > 0 {
-		if message.Len() > 0 {
-			message.WriteString(" and ")
+		if response.EntityCount.CompCount > 0 {
+			componentWord := determinePluralWord(response.EntityCount.CompCount, "component")
+			message.WriteString(fmt.Sprintf("%d %s", response.EntityCount.CompCount, componentWord))
+		}
+		if response.EntityCount.RelCount > 0 {
+			if response.EntityCount.CompCount > 0 {
+				message.WriteString(" and ")
+			}
+			relationshipWord := determinePluralWord(response.EntityCount.RelCount, "relationship")
+			message.WriteString(fmt.Sprintf("%d %s", response.EntityCount.RelCount, relationshipWord))
 		}
-		relationshipWord := determinePluralWord(response.EntityCount.RelCount, "relationship")
-		message.WriteString(fmt.Sprintf("%d %s", response.EntityCount.RelCount, relationshipWord))
-	}
-	if response.EntityCount.CompCount == 0 && response.EntityCount.RelCount > 0 {
-		relationshipWord := determinePluralWord(response.EntityCount.RelCount, "relationship")
-		message.WriteString(fmt.Sprintf("%d %s", response.EntityCount.RelCount, relationshipWord))
-	}
-	if response.EntityCount.CompCount > 0 || response.EntityCount.RelCount > 0 {
 		message.WriteString(")")
 	}
+
 	return message.String()
 }
 
 func ErrMsgContruct(response *models.RegistryAPIResponse) string {
 	entityWord := determinePluralWord(response.EntityCount.TotalErrCount, "entity")
-
 	msg := fmt.Sprintf("encountered error for %d %s (", response.EntityCount.TotalErrCount, entityWord)
-	componentsPresent := response.EntityCount.ErrCompCount > 0
-	relationshipsPresent := response.EntityCount.ErrRelCount > 0
-	modelsPresent := response.EntityCount.ErrModelCount > 0
-	knownErrors := response.EntityCount.ErrCompCount + response.EntityCount.ErrRelCount + response.EntityCount.ErrModelCount
-	unknownErrors := response.EntityCount.TotalErrCount - knownErrors
 
-	// Collect errors in a slice for dynamic message construction
 	errors := []string{}
-	if modelsPresent {
+	if response.EntityCount.ErrModelCount > 0 {
```

Comment:
i think keeping these as variables is better for readabilty and debugging . there is no need for this .